erpnext/templates/includes/integrations/authorizenet_checkout.js

```diff
@@ -1,87 +1,308 @@
-	}
+/**
+* @typedef {Object} ProcessStatus
+* @property {string} status Status enum. Possible values: "Completed", "NotFound", "Queued", "Failed".
+* @property {string} type On a failed request. The type is one of: "Validation", "HardError", "Timeout", "Unhandled"
+* @property {string} description A reason message for the status returned.
+* @property {int} wait An amount of time the client should wait before issuing 
+*                      another status check request in milliseconds.
+*/
+
+frappe.ready(function() {
+	const data = context.replace(/'/g, '"');
+
+	/**
+	 * Fetches payment request status information
+	 * @param {*} data Order information data structure. See authorizenet_settings.py
+	 * @returns Promise<void>
+	 */
+	const fetch_status = async function(data) {
+		const r = await frappe.call({
+			method: "erpnext.erpnext_integrations.doctype.authorizenet_settings.authorizenet_settings.get_status",
+			freeze: true,
+			args: {
+				"data": data
+			}
+		});
 
-	if (!isValidCard.isPotentiallyValid) {
-		frappe.throw(__("Card Number is Invalid."));
+		return r.message;
 	}
 
-	if(cardNumber.length < 13 || cardNumber.length > 16){
-		frappe.throw(__("Card Number length should be between 13 and 16 characters"));
+	/**
+	 * Starts status request and UI updates to handle PR status check.
+	 * @param {object} data Order information data structure. See authorizenet_settings.py
+	 * @param {string} label Status label to display when processing status
+	 * @param {onCardProcessSuccess} on_success a Success function callback. 
+	 * @param {onCardProcessFail} on_fail A Failure function callback
+	 * @param {int} counter A counter which is incremented internally as timeouts occur
+	 * @param {int} maxTimeouts Maximum number of timeouts before a hard error
+	 * @returns Promise<void>
+	 */
+	const check_status = async function(data, label, on_success, on_fail, counter, maxTimeouts) {
+		if ( !maxTimeouts) {
+			maxTimeouts = 6; // 6 times x 5 secs = 30secs
+		}
+		// switch ui to display message to user
+		start_status_check(label);
+		try {
+			const status = await fetch_status(data);
+
+			// Process status result
+			await process_status(status, label, on_success, on_fail, counter, maxTimeouts);
+		} catch (err) {
+			if ( err ) {
+				if ( !navigator.onLine || err.status === 0 ) {
+					if ( counter < maxTimeouts ) {
+						setTimeout(() => check_status(data, 
+							"Detected a network problem. Trying to reconnect, please wait...",
+							on_success, 
+							on_fail, 
+							counter+1), 
+							5000, 
+							maxTimeouts
+						);
+						return; // only early exit when queuing up a delayed check_status.
+					} else if ( on_fail ) {
+						on_fail({
+							"status": "Failed",
+							"type": "HardError",
+							"description": "Network problem! Unfortunately we could to connect with our server. If your internet is down please try again later. Otherwise, contact support to help with your transction."
```

Comment:
```suggestion
							"description": "Network problem! Unfortunately we could not connect to our server. If your internet is down please try again later. Otherwise, contact support to help with your transaction."
```